crates/nu-command/src/dataframe/lazy/fill_nan.rs

```diff
@@ -0,0 +1,125 @@
+use crate::dataframe::values::{Column, NuDataFrame, NuExpression, NuLazyFrame};
+use nu_engine::CallExt;
+use nu_protocol::{
+    ast::Call,
+    engine::{Command, EngineState, Stack},
+    Category, Example, PipelineData, ShellError, Signature, Span, SyntaxShape, Type, Value,
+};
+use polars::datatypes::DataType;
+
+#[derive(Clone)]
+pub struct LazyFillNA;
+
+impl Command for LazyFillNA {
+    fn name(&self) -> &str {
+        "fill-nan"
+    }
+
+    fn usage(&self) -> &str {
+        "Replaces NA values with the given expression"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build(self.name())
+            .required(
+                "fill",
+                SyntaxShape::Any,
+                "Expression to use to fill the NAN values",
+            )
+            .input_type(Type::Custom("dataframe".into()))
+            .output_type(Type::Custom("dataframe".into()))
+            .category(Category::Custom("lazyframe".into()))
+    }
+
+    fn examples(&self) -> Vec<Example> {
+        vec![Example {
+            description: "Fills the NaN values by 0",
+            example: "[1 2 NaN 3 NaN] | into df | fill-nan 0",
+            result: Some(
+                NuDataFrame::try_from_columns(vec![Column::new(
+                    "0".to_string(),
+                    vec![
+                        Value::test_int(1),
+                        Value::test_int(2),
+                        Value::test_int(0),
+                        Value::test_int(3),
+                        Value::test_int(0),
+                    ],
+                )])
+                .expect("Df for test should not fail")
+                .into_value(Span::test_data()),
+            ),
+        }]
+    }
+
+    fn run(
+        &self,
+        engine_state: &EngineState,
+        stack: &mut Stack,
+        call: &Call,
+        input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        let fill: Value = call.req(engine_state, stack, 0)?;
+        let value = input.into_value(call.head);
+
+        if NuExpression::can_downcast(&value) {
+            let expr = NuExpression::try_from_value(value)?;
+            let fill = NuExpression::try_from_value(fill)?.into_polars();
+            let expr: NuExpression = expr.into_polars().fill_nan(fill).into();
+
+            Ok(PipelineData::Value(
+                NuExpression::into_value(expr, call.head),
+                None,
+            ))
+        } else {
+            let cloned = value.clone();
+            let span = cloned.span()?;
+            let _type = NuDataFrame::get_df(cloned)?.get_type();
+
+            if _type[0] == DataType::Float64 {
```

Comment:
This type check only works for single columns correctly and else will only look at the first column, which is a problem for full dataframes if the first column is a float:
![grafik](https://user-images.githubusercontent.com/15833959/209410739-ca996884-b3cb-4d66-a0b5-d7736926e5ed.png)

**Example 1:** non numeric data in the second column. (I don't think we have to worry about that too much if we consider that outside the scope for `fill-nan`)

**Example 2:** same problem as reported in #6888 for a single column of mixed ints and nan.

**Example 3:** works because everything was valid float data before `into df`

I think to close #6888 example 2 should also work.

**NB:** `_variable` names by convention are mostly used in special circumstances (unused params and macro interior). To avoid keyword conflicts there is the `r#` prefix but in this case it is probably better to just give a slightly longer more descriptive name.

Reply:
I solved this issue .
Yes `_type` parameter actually I used is force code to act code like this. 
Beside of this issue , do you think we have to implement something which is going to be use with this example too  ? 
`[[a,b]; [[nan nan] [nan nan]] | fill-nan 0 ]`

I can implement something in `nu_dataframe` to call itself and match types to replace NaNs values .

Do you think is it a good id ?

Reply:
Thanks for your time and helping and apologize to not been patient , I am not familiar very lot yet , and new on this platform ( on contributing ) .
Anyway I deleted our first condition , because our second can do it anyway .
I added for List and can be processed now with Dataframe in Dataframe too now .

Thanks again @sholderbach  . Learning from you .